packages/test-project/tests/ledger-e2e.test.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * Basic e2e tests for ensuring functionality of Ledger devices
+ *
+ * @group ledger
+ */
+
+const { wallet } = require('./common/constants');
+const {
+  openExtension,
+  getOpenedTab,
+  selectAccount,
+  verifyUITransaction,
+  getLedgerSuggestedParams,
+  decodeBase64Blob,
+  buildSdkTx,
+  prepareWalletTx,
+  sendTransaction,
+} = require('./common/helpers');
+const {
+  CreateWallet,
+  VerifyAccount,
+  ConnectAlgoSigner,
+} = require('./common/tests');
+
+jest.setTimeout(15000);
+jest.retryTimes(0, { logErrorsBeforeRetry: true });
+const testAssetIndex = 13169404;
+const linkedLedgerAccount = {
+  name: 'Ledger Account',
+};
+let openedTab;
+let ledgerParams;
+
+async function signDappTxn(transactionsToSign) {
+  return await dappPage.evaluate(async (transactionsToSign) => {
+    const signPromise = AlgoSigner.signTxn(transactionsToSign)
+      .then((data) => {
+        return data;
+      })
+      .catch((error) => {
+        return error;
+      });
+    await window['authorizeSignTxn']();
+    return signPromise;
+  }, transactionsToSign);
+}
+
+describe('Wallet Setup', () => {
+  beforeAll(async () => {
+    await openExtension();
+  });
+
+  CreateWallet();
+  ConnectAlgoSigner();
+
+  test('Get TestNet params', async () => {
+    ledgerParams = await getLedgerSuggestedParams();
+  });
+});
+
+// Create a new account in AlgoSigner
+describe('Link Ledger Account', () => {
+  test('Link An Account, Step 1 - Begin Linking Process', async () => {
+    await extensionPage.waitForSelector('#addAccount');
+    await extensionPage.click('#addAccount');
+    await extensionPage.click('#linkAccount');
+    await extensionPage.click('#nextStep');
+    openedTab = await getOpenedTab();
+  });
+
+  test('Link An Account, Step 2 - Select Address to Link and Name', async () => {
+    await openedTab.waitForSelector('#loadAccounts');
+    await openedTab.click('#loadAccounts');
+    await openedTab.waitForSelector('[data-public-address]');
+    linkedLedgerAccount.address = await openedTab.$eval(
+      '[data-public-address]',
+      (e) => e.innerText
+    );
+    await openedTab.type('#accountName', linkedLedgerAccount.name);
+    await openedTab.click('#nextStep');
+  });
+
+  test('Link an Account, Step 3 - Write Account into Storage', async () => {
+    await openedTab.waitForSelector('#enterPassword');
+    await openedTab.type('#enterPassword', wallet.password);
+    await openedTab.waitForTimeout(200);
+    await openedTab.click('#authButton');
+  });
+
+  test('Link An Account, Step 4 - Close tab and reload extension', async () => {
+    await openedTab.waitForTimeout(2000);
```

Comment:
This test was failing with 2000. Seems like it opened the extension too fast and the account wasn't there yet:
![image](https://user-images.githubusercontent.com/52399794/188854740-8769a18f-7064-4583-b986-422ec554563d.png)
Increasing it to 4000 worked for me. Maybe Linux is slower?